# Simpson's Paradox
Use `admission_data.csv` for this exercise.

No ejecutar ninguna celda

In [2]:
import pandas as pd

In [3]:
# Load and view first few lines of dataset
df = pd.read_csv('admission_data.csv')
df.head()

student_id  gender      major  admitted
0       35377  female  Chemistry     False
1       56105    male    Physics      True
2       31441  female  Chemistry     False
3       51765    male    Physics      True
4       53714  female    Physics      True

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 500 entries, 0 to 499
Data columns (total 4 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   student_id  500 non-null    int64 
 1   gender      500 non-null    object
 2   major       500 non-null    object
 3   admitted    500 non-null    bool  
dtypes: bool(1), int64(1), object(2)
memory usage: 12.3+ KB


### Proportion and admission rate for each gender

In [9]:
# get number of males and females in list
 
#get total number of values in gender column in variable


In [5]:
df.gender.value_counts()

female    257
male      243
Name: gender, dtype: int64

In [9]:
# Proportion of students that are female

print('Proportion of females is ' + str(df.gender.value_counts()['female']/len(df) * 100) +'%')
print('Proportion of females is ' + str(df.gender.value_counts()['female']/len(df)))

Proportion of females is 51.4%
Proportion of females is 0.514


In [10]:
# Proportion of students that are male
print('Proportion of males is ' + str(df.gender.value_counts()['male']/len(df) * 100) +'%')
print('Proportion of males is ' + str(df.gender.value_counts()['male']/len(df)))

Proportion of males is 48.6%
Proportion of males is 0.486


In [11]:
#To view statistics based on gender, we group by gender variable
# Next we use value_counts() function to see the numbers
df[['gender', 'admitted']].value_counts()

gender  admitted
female  False       183
male    False       125
        True        118
female  True         74
dtype: int64

In [12]:
# count of females
# https://chrisalbon.com/python/pandas_selecting_rows_on_conditions.html

print('Number of female students are ' + str(len(df[df['gender'] == 'female'])))

Number of female students are 257


In [13]:
#count of males
print('Number of male students are ' + str(len(df[df['gender'] == 'male'])))

Number of male students are 243


In [14]:
# Admission rate for females

import numpy as np
ad_rate_fem = len(df[(df['gender'] == 'female') & (df['admitted'] == True)])/len(df[df['gender'] == 'female'])
print('Female admission rate is ' + str(np.round(ad_rate_fem*100,2)) + '%')

Female admission rate is 28.79%


In [15]:
# Admission rate for males

ad_rate_ma = len(df[(df['gender'] == 'male') & (df['admitted'] == True)])/len(df[df['gender'] == 'male'])
print('Male admission rate is ' + str(np.round(ad_rate_ma*100,2)) + '%')

Male admission rate is 48.56%


In [16]:
print('Total admission proportions')
print('Female: ' + str(ad_rate_fem))
print('Male: ' + str(ad_rate_ma))

Total admission proportions
Female: 0.28793774319066145
Male: 0.48559670781893005


### Proportion and admission rate for physics majors of each gender

In [17]:
# group by major and gender to get counts of students in each cohort
df.groupby(['major', 'gender']).count()

student_id  admitted
major     gender                      
Chemistry female         226       226
          male            18        18
Physics   female          31        31
          male           225       225

In [18]:
# What proportion of female students are majoring in physics?

# First, get count of females in Physics major 


#now evaluate proportion of female candidates in Phsics
fe_phy = len(df[(df['gender'] == 'female') & (df['major'] == 'Physics')])
print('Number of females with Physics major is ' + str(fe_phy))

ad_rate_fem_can_phy = len(df[(df['gender'] == 'female') & (df['major'] == 'Physics')])/len(df[df['gender'] == 'female'])
print('Proportion of female candidates in Physics is ' + str(np.round(ad_rate_fem_can_phy*100,2)) + '%')


Number of females with Physics major is 31
Proportion of female candidates in Physics is 12.06%


In [19]:
# What proportion of male students are majoring in physics?

# first,get count of males in Physics major 


#now evaluate proportion of male candidates in Physics
ma_phy = len(df[(df['gender'] == 'male') & (df['major'] == 'Physics')])
print('Number of males with Physics major is ' + str(ma_phy))

ad_rate_ma_can_phy = len(df[(df['gender'] == 'male') & (df['major'] == 'Physics')])/len(df[df['gender'] == 'male'])
print('Proportion of male candidates in Physics is ' + str(np.round(ad_rate_ma_can_phy*100,2)) + '%')

Number of males with Physics major is 225
Proportion of male candidates in Physics is 92.59%


In [20]:
# Admission rate for female physics majors
# compute count of females in Physics who got admitted


#now evaluate proportion of admitted female candidates in Phsics
ad_rate_fem_ad_phy = len(df[(df['gender'] == 'female') & (df['admitted'] == True) & (df['major'] == 'Physics')])/len(df[(df['gender'] == 'female') & (df['major']=='Physics') ])
print('Proportion of female candidates admitted in Physics is ' + str(np.round(ad_rate_fem_ad_phy*100,2)) + '%')

Proportion of female candidates admitted in Physics is 74.19%


In [21]:
# Admission rate for male physics majors
# compute count of females in Physics who got admitted


#now evaluate proportion of admitted male candidates in Phsics
ad_rate_ma_ad_phy = len(df[(df['gender'] == 'male') & (df['admitted'] == True) & (df['major'] == 'Physics')])/len(df[(df['gender'] == 'male') & (df['major']=='Physics') ])
print('Proportion of male candidates admitted in Physics is ' + str(np.round(ad_rate_ma_ad_phy*100,2)) + '%')

Proportion of male candidates admitted in Physics is 51.56%


### Proportion and admission rate for chemistry majors of each gender

In [22]:
# What proportion of female students are majoring in chemistry?

# First, get count of females in Chemistry major 


#now evaluate proportion of female candidates in Chemistry
fe_che = len(df[(df['gender'] == 'female') & (df['major'] == 'Chemistry')])
print('Number of females with Chemistry major is ' + str(fe_che))

ad_rate_fem_can_che = len(df[(df['gender'] == 'female') & (df['major'] == 'Chemistry')])/len(df[df['gender'] == 'female'])
print('Proportion of female candidates in Chemistry is ' + str(np.round(ad_rate_fem_can_che*100,2)) + '%')

Number of females with Chemistry major is 226
Proportion of female candidates in Chemistry is 87.94%


In [23]:
# What proportion of male students are majoring in chemistry?
# First, get count of females in Chemistry major 


#now evaluate proportion of male candidates in Chemistry
ma_che = len(df[(df['gender'] == 'male') & (df['major'] == 'Chemistry')])
print('Number of males with Chemistry major is ' + str(ma_che))

ad_rate_ma_can_che = len(df[(df['gender'] == 'male') & (df['major'] == 'Chemistry')])/len(df[df['gender'] == 'male'])
print('Proportion of male candidates in Chemistry is ' + str(np.round(ad_rate_ma_can_che*100,2)) + '%')

Number of males with Chemistry major is 18
Proportion of male candidates in Chemistry is 7.41%


In [24]:
# Admission rate for female chemistry majors

# compute count of females in chemistry who got admitted


#now evaluate proportion of admitted female candidates in chemistry
ad_rate_fem_ad_che = len(df[(df['gender'] == 'female') & (df['admitted'] == True) & (df['major'] == 'Chemistry')])/len(df[(df['gender'] == 'female') & (df['major']=='Chemistry') ])
print('Proportion of female candidates admitted in Chemistry is ' + str(np.round(ad_rate_fem_ad_che*100,2)) + '%')

Proportion of female candidates admitted in Chemistry is 22.57%


In [25]:
# Admission rate for male chemistry majors
# compute count of males in chemistry who got admitted


#now evaluate proportion of admitted male candidates in chemistry
ad_rate_ad_phy = len(df[(df['admitted'] == True) & (df['major'] == 'Physics')])/len(df[(df['major']=='Physics')])
print('Proportion of candidates admitted in Physics is ' + str(np.round(ad_rate_ad_phy*100,2)) + '%')

Proportion of candidates admitted in Physics is 54.3%


### Admission rate for each major

In [26]:
# Admission rate for physics majors
# compute count of students in Physics who got admitted


#now evaluate proportion of admitted candidates in Phsics
ad_rate_ad_che = len(df[(df['admitted'] == True) & (df['major'] == 'Chemistry')])/len(df[(df['major']=='Chemistry')])
print('Proportion of candidates admitted in Chemistry is ' + str(np.round(ad_rate_ad_che*100,2)) + '%')

Proportion of candidates admitted in Chemistry is 21.72%


In [29]:
# Admission rate for chemistry majors
# compute count of students in Chemistry who got admitted


#now evaluate proportion of admitted candidates in Chemistry
    
ad_rate_not_ad_che = len(df[(df['admitted'] == False) & (df['major'] == 'Chemistry')])/len(df[(df['major']=='Chemistry')])
if (ad_rate_not_ad_che + ad_rate_ad_che) == 1:
    print('Values computed for Chemistry admitted candidates is Correct')

Values computed for Chemistry admitted candidates is Correct


#### Testing: Double check vlaues by computing non-admittance values

In [30]:
# double check values for physics admittance



# double check values for Chemistry admittance
ad_rate_not_ad_phy = len(df[(df['admitted'] == False) & (df['major'] == 'Physics')])/len(df[(df['major']=='Physics')])
# print('Proportion of candidates not admitted in Physics is ' + str(np.round(ad_rate_not_ad_phy*100,2)) + '%')
if (ad_rate_not_ad_phy + ad_rate_ad_phy) == 1:
    print('Values computed for Physics admitted candidates is Correct')

Values computed for Physics admitted candidates is Correct


In [32]:
# double check not admitted candidates
# candidates not admitted to Physics

# candidates not admitted to Chemistry

# compare if admitted and non-admitted candidate counts equal total number of students


Values computed for admitted candidates is Correct


Proportion of females is 51.4%
Proportion of males is 48.6%

Mitad de hombres y de mujeres

Female admission rate is 28.79%
Male admission rate is 48.56%

Admiten al 50% de hombres y al 30% de mujeres, ¿hay discriminación?

Proportion of female candidates in Physics is 12.06%
Proportion of male candidates in Physics is 92.59%

Hay más hombres intentando entrar en Física que mujeres (en proporción)

Proportion of female candidates admitted in Physics is 74.19%
Proportion of male candidates admitted in Physics is 51.56%

Hay más proporción de mujeres admitidas en Física que hombres admitidos (en proporción)

Proportion of female candidates in Chemistry is 87.94%
Proportion of male candidates in Chemistry is 7.41%

Hay más mujeres intentando entrar en Química que hombres (en proporción)

Proportion of female candidates admitted in Chemistry is 22.57%
Proportion of male candidates admitted in Chemistry is 11.11%

Hay más mujeres aceptadas en Química que hombres (en proporción)

Proportion of candidates admitted in Physics is 54.3%
Proportion of candidates admitted in Chemistry is 21.72%

Se admite al 54% de candidatos en Física y al 22% de candidatos en Química.

¿Hay discriminación?

Los candidatos son la mitad hombres y la mitad mujeres.
Admiten al 50% de hombres y al 30% de mujeres. Nota 1
Los hombres intentan entrar en Física y las mujeres en Química.
Hay más proporción de mujeres aceptadas tanto en Física como en Química. Nota 2
¿Hay discriminación?
Es más fácil entrar en Física que en Química.

Las notas 1 y 2 son un ejemplo de paradoja de Simpsons.

This is also the explanation of why the overall acceptance rate for women is lower than the rate for men. It’s not that women were discriminated against by any departments (at least as far as we know!), it’s that women — in comparison to men — seemed to apply more to very competitive departments where it was really hard to get in.
